app/components/UserList/UserListItem.tsx

```diff
@@ -9,120 +9,142 @@ interface UserListItemProps {
   username: string;
   isSelf: boolean | undefined;
   isDJ: boolean | undefined;
+  currentMiniRoom: string;
+  inConference: boolean;
 }
 
-const UserListItem = memo(
-  ({ userMachineId, username, isSelf, isDJ }: UserListItemProps) => {
-    const dispatch = useDispatch();
-    const audioRef = useRef();
-    const [isEditing, setIsEditing] = useState(false);
-    const [inputName, setInputName] = useState(username);
+const UserListItem = ({
+  userMachineId,
+  username,
+  isSelf,
+  isDJ,
+  currentMiniRoom,
+  inConference,
+}: UserListItemProps) => {
+  const dispatch = useDispatch();
+  const audioRef = useRef<HTMLAudioElement>();
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputName, setInputName] = useState(username);
 
-    const conferenceUser = useSelector((state) => {
-      const { jitsiParticipants } = conferenceSelector(state);
-      const user = _.find(
-        jitsiParticipants,
-        ({ machineId }) => machineId === userMachineId
-      );
+  const conferenceUser = useSelector((state) => {
+    const { jitsiParticipants } = conferenceSelector(state);
+    return _.find(
+      jitsiParticipants,
+      ({ machineId }) => machineId === userMachineId
+    );
+  });
 
-      return user;
-    });
+  useEffect(() => {
+    if (!conferenceUser || isSelf || !conferenceUser.hasTrack) {
+      return;
+    }
 
-    useEffect(() => {
-      if (!conferenceUser || isSelf || !conferenceUser.hasTrack) {
-        return;
-      }
+    const remoteTrack = _.get(window, [
+      'JitsiRemoteTracks',
+      conferenceUser.conferenceId,
+    ]);
 
-      const remoteTrack = _.get(window, [
-        'JitsiRemoteTracks',
-        conferenceUser.conferenceId,
-      ]);
+    if (remoteTrack && _.isEmpty(_.get(remoteTrack, 'containers'))) {
+      // eslint-disable-next-line no-console
+      console.error('EFFECT CONFERENCE', conferenceUser, remoteTrack);
+      remoteTrack.attach(audioRef.current);
+    }
+  }, [isSelf, conferenceUser]);
 
-      if (remoteTrack && _.isEmpty(_.get(remoteTrack, 'containers'))) {
-        // eslint-disable-next-line no-console
-        console.error('EFFECT CONFERENCE', conferenceUser, remoteTrack);
-        remoteTrack.attach(audioRef.current);
-      }
-    }, [isSelf, conferenceUser]);
+  useEffect(() => {
+    setInputName(username);
+  }, [username]);
 
-    useEffect(() => {
-      setInputName(username);
-    }, [username]);
+  // This effect will mute participants based on the miniroom
+  useEffect(() => {
+    console.error(conferenceUser, currentMiniRoom, audioRef.current);
+    if (!conferenceUser || !conferenceUser.miniRoom || !audioRef.current)
+      return;
```

Comment:
```suggestion
    if (!conferenceUser || !conferenceUser.miniRoom || !audioRef.current) {
      return;
    }
    
```

I like for readability reasons to keep if blocks wrapped with {}